samcli/lib/config/samconfig.py

```diff
@@ -70,22 +81,21 @@ def get_all(self, cmd_names, section, env=DEFAULT_ENV):
         KeyError
             If the config file does *not* have the specific section
 
-        tomlkit.exceptions.TOMLKitError
+        SamConfigFileReadException
             If the configuration file is invalid
         """
 
         env = env or DEFAULT_ENV
 
-        self._read()
-        if isinstance(self.document, dict):
-            toml_content = self.document.get(env, {})
-            params = toml_content.get(self._to_key(cmd_names), {}).get(section, {})
-            if DEFAULT_GLOBAL_CMDNAME in toml_content:
-                global_params = toml_content.get(DEFAULT_GLOBAL_CMDNAME, {}).get(section, {})
-                global_params.update(params.copy())
-                params = global_params.copy()
-            return params
-        return {}
+        self.document = self.file_manager.read(self.filepath)
```

Comment:
Any reason why `self._read()` is not called here directly?

Reply:
That's a good catch, calling `self._read()` would also probably be faster here since `self.document` would likely already be stored and we wouldn't have to do any I/O. I'll update accordingly.